### PR TITLE
[#809] Reduce code duplication in Xbase tests

### PR DESCRIPTION
**Pull request by szarnekow**
Also fix package names in purexbase.tests

Signed-off-by: Sebastian Zarnekow <sebastian.zarnekow@gmail.com>

### CODE REVIEW

File: org.eclipse.xtext.purexbase.tests/src/org/eclipse/xtext/purexbase/tests/ParserTest.java [hunk 1]
```diff
@@ -6,12 +6,11 @@
  *
  * SPDX-License-Identifier: EPL-2.0
  *******************************************************************************/
-package org.eclipse.xtext.purexbase.test;
```
Comment (cdietrich) [inline org.eclipse.xtext.purexbase.tests/src/org/eclipse/xtext/purexbase/tests/ParserTest.java:9]: does this survive gradlew generatePureXbase?

**Comment by cdietrich** [comment]
can you check the test fails

File: org.eclipse.xtext.purexbase.tests/src/org/eclipse/xtext/purexbase/tests/ParserTest.java (hunk 1 above)
Comment (cdietrich) [reply org.eclipse.xtext.purexbase.tests/src/org/eclipse/xtext/purexbase/tests/ParserTest.java:9]: generate stub is false

File: org.eclipse.xtext.purexbase.tests/src/org/eclipse/xtext/purexbase/tests/RuntimeInjectorProvider.java [hunk 2]
```diff
@@ -6,10 +6,9 @@
  *
  * SPDX-License-Identifier: EPL-2.0
  *******************************************************************************/
-package org.eclipse.xtext.purexbase.test;
+package org.eclipse.xtext.purexbase.tests;
 
 import org.eclipse.xtext.purexbase.PureXbaseStandaloneSetup;
-import org.eclipse.xtext.purexbase.tests.PureXbaseInjectorProvider;
```
[the remark this answers is not in the document]
Comment (cdietrich) [reply org.eclipse.xtext.purexbase.tests/src/org/eclipse/xtext/purexbase/tests/RuntimeInjectorProvider.java:12]: got it, was always package tests, now same package, no issue

**Comment by cdietrich** [comment]
tests still seems red

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/AbstractXbaseCompiler.java [hunk 3]
```diff
@@ -35,10 +35,12 @@
 import org.eclipse.xtext.common.types.util.Primitives.Primitive;
 import org.eclipse.xtext.util.Strings;
 import org.eclipse.xtext.xbase.XAbstractFeatureCall;
+import org.eclipse.xtext.xbase.XBinaryOperation;
```
Comment (szarnekow) [inline org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/AbstractXbaseCompiler.java:38]: @cdietrich The package rename revealed more NPEs with broken models.

I added more safeguards below which you may want to double-check.

**Review by cdietrich — APPROVED** [review]

**Pull request merged**